src/ServerTelemetryChannel/Implementation/ApplicationFolderProvider.cs

```diff
@@ -22,12 +22,21 @@ internal class ApplicationFolderProvider : IApplicationFolderProvider
         private readonly IIdentityProvider identityProvider;
 
         public ApplicationFolderProvider(string folderName = null)
-            : this(Environment.GetEnvironmentVariables(), folderName)
+            : this(null, folderName)
         {
         }
 
         internal ApplicationFolderProvider(IDictionary environment, string folderName = null)
         {
+            try
+            {
+                environment = Environment.GetEnvironmentVariables();
+            }
+            catch (SecurityException e)
+            {
+                throw new ArgumentNullException(nameof(environment), e);
```

Comment:
this doesn't look right to me.
If GetEnvironmentVariables throws a [SecurityException](https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariables?view=netframework-4.7.2) that would be a separate problem than an ArgumentNull